### ISSUE TITLE
no-code-outside-tag

### ISSUE THREAD

**Issue by tjx666** [vuejs/eslint-plugin-vue#2209]
**Please describe what the rule should do:**

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<template>
  <div>666</div>
</template>

console.log(1);

<script>
const num = 1;
console.log(num);
</script>

```

**Additional context**

the statement `console.log(1);` outside the tag will be ignored by vue compiler, and not work as expected

**Comment by FloEdelmann** [comment]
Hmm, is this a common problem? Doesn't the syntax highlighting in the editor already tell that it is in the wrong place?

But I guess this rule could be added to eslint-plugin-vue. Pull requests welcome! Note that it should not report comments (`<!-- … -->`) or custom blocks (e.g. `<docs>…</docs>`), but all text content (e.g. `hello world`, not just valid JS code like in the example above).

**Comment by ValentinYoushkevich** [comment]
Opened #3119. Comments and custom blocks are left alone as you asked, and removing the stray content is offered as a suggestion rather than an autofix.
